fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/mixin/recipe/cooking/AbstractFurnaceBlockEntityMixin.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.recipe.cooking;
+
+import com.llamalad7.mixinextras.injector.ModifyExpressionValue;
+import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
+import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
+import com.llamalad7.mixinextras.sugar.Local;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+
+import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.recipe.RecipeEntry;
+import net.minecraft.registry.DynamicRegistryManager;
+
+/**
+ * A Mixin class to ensure that furnaces can use the functionality provided from recipes involving multiple output items
+ * (not item types, just multiple of the same item).
+ */
+@Mixin(AbstractFurnaceBlockEntity.class)
+public abstract class AbstractFurnaceBlockEntityMixin {
+	/**
+	 * A wrapper altering the result of a smelting recipe when outputting into a partially filled output slot. 1.20.4
+	 * functionality increments the output by 1. This wrapper instead increases the count by the amount defined in the
+	 * recipe.
+	 *
+	 * <p>Assumes that the increment will be valid and does not stop recipe if it would go over. E.g. an increment of
+	 * 65 would always be performed, if not stopped by other logic prior to this.
+	 *
+	 * @param outputStack the instance of the itemstack to increment, in the furnace's output slot.
+	 * @param increment the original incremented amount (1.20.4 this is always 1).
+	 * @param incrementItemStack the method responsible for increasing the item stack count.
+	 * @param registryManager the registry manager to access for the recipe
+	 * @param recipe the (!= null) recipe.
+	 */
+	@WrapOperation(
+			method = "craftRecipe",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/item/ItemStack;increment(I)V"
+			)
+	)
+	private static void incrementByRecipe(ItemStack outputStack, int increment, Operation<Void> incrementItemStack, DynamicRegistryManager registryManager, RecipeEntry<?> recipe) {
```

Comment:
Should be a `@ModifyArg` with `@Local(argsOnly = true)`s to get the enclosing params.